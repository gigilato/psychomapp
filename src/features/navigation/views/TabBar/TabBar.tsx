import { BottomTabBarProps } from '@react-navigation/bottom-tabs'

import { HStack } from '$atoms'
import { TabBarIcon } from '$features/navigation/views/TabBar/components'
import { i18n } from '$infra/i18n'
import { useSafeAreaInsets } from '$views/libs/safeAreaInsets'

export const TabBar = ({ state: { index } }: BottomTabBarProps) => {
  const { bottom } = useSafeAreaInsets()
  return (
    <HStack
      shadow
      className="absolute self-center py-s px-xs rounded-xl justify-around"
      style={{ bottom }}
    >
      <TabBarIcon
        route="dashboard"
        icon="calendar"
        isSelected={index === 0}
        name={i18n.t('navigation.calendar')}
      />
      <TabBarIcon
        route="dashboard/patients"
        icon="users"
        isSelected={index === 2}
        name={i18n.t('navigation.patients')}
      />
      <TabBarIcon
        route="dashboard/profile"
        icon="sliders-horizontal"
        isSelected={index === 1}
        name={i18n.t('navigation.settings')}
      />
    </HStack>
  )
}
