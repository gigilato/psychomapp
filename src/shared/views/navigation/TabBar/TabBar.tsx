import { useLayout } from '@react-native-community/hooks'
import { BottomTabBarProps } from '@react-navigation/bottom-tabs'
import { useEffect } from 'react'
import { useAnimatedStyle } from 'react-native-reanimated'

import { AnimatedHStack } from '$atoms'
import { i18n } from '$infra/i18n'
import { setTabBarHeight, useLayoutStore } from '$infra/layout'
import { useTransition, mix } from '$views/libs/animations'
import { useSafeAreaInsets } from '$views/libs/safeAreaInsets'

import { TabBarIcon } from './components'

export const TabBar = ({ state: { index } }: BottomTabBarProps) => {
  const { bottom } = useSafeAreaInsets()
  const isTabBarVisible = useLayoutStore((state) => state.isTabBarVisible)
  const transition = useTransition(isTabBarVisible)

  const aStyle = useAnimatedStyle(() => {
    return { bottom: mix(transition, [-200, bottom]), opacity: mix(transition, [0, 1]) }
  })

  const { onLayout, height } = useLayout()
  useEffect(() => {
    setTabBarHeight(height + bottom)
  }, [height, bottom])

  return (
    <AnimatedHStack
      shadow
      className="absolute self-center py-s px-xs rounded-xl justify-around"
      style={aStyle}
      onLayout={onLayout}
    >
      <TabBarIcon
        route="/dashboard/"
        icon="calendar"
        isSelected={index === 0}
        name={i18n.t('navigation.calendar')}
      />
      <TabBarIcon
        route="/dashboard/patients/"
        icon="users"
        isSelected={index === 2}
        name={i18n.t('navigation.patients')}
      />
      <TabBarIcon
        route="/dashboard/profile/"
        icon="settings"
        isSelected={index === 1}
        name={i18n.t('navigation.settings')}
      />
    </AnimatedHStack>
  )
}
