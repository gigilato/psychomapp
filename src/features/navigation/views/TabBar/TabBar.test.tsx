import { BottomTabBarProps } from '@react-navigation/bottom-tabs'
import { render } from '@testing-library/react-native'
import React from 'react'

import { TabBar } from './TabBar'

const mockProps: BottomTabBarProps = {
  descriptors: {},
  state: {
    index: 0,
    history: [],
    key: 'key',
    routeNames: ['routeNames'],
    routes: [],
    stale: false,
    type: 'tab',
  },
  navigation: {
    navigate: jest.fn(),
    dispatch: jest.fn(),
    reset: jest.fn(),
    goBack: jest.fn(),
    isFocused: jest.fn(),
    canGoBack: jest.fn(),
    getId: jest.fn(),
    getState: jest.fn(),
    getParent: jest.fn(),
    setParams: jest.fn(),
    emit: jest.fn(),
  },
  insets: { bottom: 0, left: 0, right: 0, top: 0 },
}

describe('TabBar', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<TabBar {...mockProps} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
