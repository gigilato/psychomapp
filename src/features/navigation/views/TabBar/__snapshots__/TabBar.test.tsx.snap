// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFCF9",
      },
      {
        "shadowColor": "rgba(0, 0, 0, 0.1)",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 6,
      },
      {
        "shadowColor": "#D3D3D3",
      },
      {
        "flexDirection": "row",
      },
      {
        "position": "absolute",
      },
      {
        "alignSelf": "center",
      },
      {
        "paddingBottom": 12,
        "paddingTop": 12,
      },
      {
        "paddingLeft": 8,
        "paddingRight": 8,
      },
      {
        "borderBottomLeftRadius": 24,
        "borderBottomRightRadius": 24,
        "borderTopLeftRadius": 24,
        "borderTopRightRadius": 24,
      },
      {
        "justifyContent": "space-around",
      },
      [
        undefined,
        {
          "bottom": 16,
        },
      ],
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
        },
        {
          "width": 90,
        },
        [
          {
            "opacity": undefined,
          },
          undefined,
        ],
      ]
    }
    testID="TabBarIcon/Agenda"
  >
    <Text />
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-Regular",
          },
          {
            "fontSize": 16,
          },
          {
            "lineHeight": 18,
          },
          {
            "color": "#FFB300",
          },
        ]
      }
    >
      Agenda
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
        },
        {
          "width": 90,
        },
        [
          {
            "opacity": undefined,
          },
          undefined,
        ],
      ]
    }
    testID="TabBarIcon/Patients"
  >
    <Text />
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-Regular",
          },
          {
            "fontSize": 16,
          },
          {
            "lineHeight": 18,
          },
        ]
      }
    >
      Patients
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
        },
        {
          "width": 90,
        },
        [
          {
            "opacity": undefined,
          },
          undefined,
        ],
      ]
    }
    testID="TabBarIcon/Menu"
  >
    <Text />
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-Regular",
          },
          {
            "fontSize": 16,
          },
          {
            "lineHeight": 18,
          },
        ]
      }
    >
      Menu
    </Text>
  </View>
</View>
`;
